src/components/time-picker/utils.ts

```diff
@@ -0,0 +1,79 @@
+export enum Ampm {
+    AM,
+    PM,
+    NONE
+}
+export type TimeSegment = 'hh' | 'mm';
+export type Format = 'ampm' | '24h';
+export type Segment = TimeSegment | 'ampm';
+
+export const isTouch = (() => {
+    if ((typeof window !== 'object') || !('orientation' in window)) {
+        return false;
+    }
+    const input = document.createElement('input');
+    input.type = 'time';
+    return input.type === 'time';
```

Comment:
* it is not part of `isTouch`, here you're checking timepicker support, it should be other function.
* `isTouch` should be part of common utils on `src` level
* also you forgot to delete element, each time you run this function new element is created, right?

is it covered with tests?